control_panel_api/permissions.py

```diff
@@ -0,0 +1,28 @@
+"""
+Custom permissions
+
+See: http://www.django-rest-framework.org/api-guide/permissions/#custom-permissions
+"""
+
+from rest_framework.permissions import BasePermission
+
+
+def is_superuser(user):
+    return user and user.is_superuser
+
+
+class IsSuperuser(BasePermission):
+    """
+    Only superusers are authorised
+    """
+
+    def has_permission(self, request, view):
+        return is_superuser(request.user)
+
+
+class AppPermissions(IsSuperuser):
```

Comment:
Rather than name the permission after the view, it would be better named to show it's intention e.g. IsAdmin or IsSUperuser - in the viewset then it would be more obvious what the class did, rather than a catch-all class where you have to access the class and look up it's hierarchy.  The rest framework has a permissions file like this.

Also I would not use empty classes for the sake of it, just use the IsSuperuser at the moment.

And at runtime can the permissions be composed of multiple classes, rather than just use one?

Reply:
I was playing with multiple permission classes and I found that if you have more than one permission class, they'll all have to return `True`.

This limits the usefulness of this mechanism, e.g. let's say we have the following viewset using these two permission classes:

```python
class SomeViewSet(viewsets.ModelViewSet):
    # ...
    permission_classes = (IsSuperuser, IsAppMaintainer,)
```

It reads better but it's not behaving as expected: both classes will need to return `True`, so you'll have to be an app maintainer and a super user at the same time, which is probably not what we'd want.

Also some of the logic may be more complicated than this and also return different things depending on the methods and/or depending on the object involved.